### PR TITLE
feat: Fixing RPC to allow bindOnIP for IPv6

### PR DESCRIPTION
**Pull request by StarAurryon**
Supporting IPv6 bindOnIP in config and RPC as discussed in #4613 .

Basically it means replacing strings package functions by net package functions.

Supporting IPv6 bindOnIP in config and RPC.

It was tested locally on a cluster with two cadence server nodes. Deployed in production with similar patch for older cadence version.

Reviewer must verify that it does not break IP address binding

IPv6 must be mentioned in changelog

bindOnIP supports IPv6 must be mentioned in doc

### CODE REVIEW

File: common/rpc/params.go
Comment (longquanzheng) [inline common/rpc/params.go:88]: Just curious, is there anything different functionally? or just a more standard way to do.

**Review by longquanzheng — APPROVED** [review]

**Review by vytautas-karpavicius — APPROVED** [review]

File: common/rpc/params.go
Comment (StarAurryon) [reply common/rpc/params.go:88]: IpV6 addresses must be represented with the port in this way [ipaddr]:port. This function does the job by ensuring that square bracket are always there if an IPv6 is provided in input.

Comment (StarAurryon) [reply common/rpc/params.go:88]: And it is of course more standard as it support both v4 and v6. Just concatenating ip:port does not work in v6 ::1:3306 is invalid as example.

**Comment by longquanzheng** [comment]
@StarAurryon can you help resolve a conflict? thanks ;)

**Comment by StarAurryon** [comment]
> @StarAurryon can you help resolve a conflict? thanks ;)

No worry, I'll do it ;)

**Comment by longquanzheng** [comment]
@StarAurryon it's possible to give us edit access to your PR branch so that we can rebase and merge later?

**Comment by longquanzheng** [comment]
Hi @StarAurryon , when you got time can you give me write access to your branch? Our repo requires rebase before merging. Without write access, I can't do rebase.

**Comment by longquanzheng** [comment]
@StarAurryon Thanks. I saw in invitation but when I clicked it it says:
```
Sorry, we couldn't find that repository invitation. It is possible that the invitation was revoked or that you are not logged into the invited account.
```

**Comment by longquanzheng** [comment]
@StarAurryon got it now. Thanks!

**Comment by StarAurryon** [comment]
It's a mess with organisational accounts... No option is shown as specified in the doc.

**Comment by longquanzheng** [comment]
> It's a mess with organisational accounts... No option is shown as specified in the doc.

No worries. Thanks for contributing. I will make a patch release today after this is merged. LMK if you have any issues when using it.

**Comment by longquanzheng** [comment]
Hi @StarAurryon 
I wasn't able to cherry-pick this commit into 0.22.x or 0.23.x release branch for v0.22.4 or 0.23.0 release. 
https://github.com/uber/cadence/tree/0.23.x
https://github.com/uber/cadence/tree/0.22.x
Because the code change you made on top of files that don't exist in any of the two branches. 

Would you mind waiting until 0.24.0 ?
If not you can make another PR for merging into any of the release branch and I will review it again. 

Thanks for the contribution again!

**Pull request merged**
